dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/geojson/job/GeoJsonImportJob.java

```diff
@@ -0,0 +1,79 @@
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.dxf2.geojson.job;
+
+import java.io.IOException;
+import java.io.InputStream;
+import lombok.AllArgsConstructor;
+import org.hisp.dhis.dxf2.geojson.GeoJsonImportReport;
+import org.hisp.dhis.dxf2.geojson.GeoJsonService;
+import org.hisp.dhis.fileresource.FileResource;
+import org.hisp.dhis.fileresource.FileResourceService;
+import org.hisp.dhis.scheduling.Job;
+import org.hisp.dhis.scheduling.JobConfiguration;
+import org.hisp.dhis.scheduling.JobProgress;
+import org.hisp.dhis.scheduling.JobType;
+import org.hisp.dhis.scheduling.parameters.GeoJsonImportJobParams;
+import org.hisp.dhis.system.notification.Notifier;
+import org.springframework.stereotype.Component;
+
+@AllArgsConstructor
+@Component
+public class GeoJsonImportJob implements Job {
+
+  private final GeoJsonService geoJsonService;
+  private final FileResourceService fileResourceService;
+  private final Notifier notifier;
+
+  @Override
+  public JobType getJobType() {
+    return JobType.GEOJSON_IMPORT;
+  }
+
+  @Override
+  public void execute(JobConfiguration jobConfig, JobProgress progress) {
+    progress.startingProcess("GeoJSON import started");
+    GeoJsonImportJobParams jobParams = (GeoJsonImportJobParams) jobConfig.getJobParameters();
+
+    progress.startingStage("Loading file resource");
+    FileResource data =
+        progress.runStage(() -> fileResourceService.getFileResource(jobConfig.getUid()));
+    progress.startingStage("Loading file content");
+
+    try (InputStream input =
+        progress.runStage(() -> fileResourceService.getFileResourceContent(data))) {
+      progress.startingStage("Importing GeoJSON started");
+      GeoJsonImportReport report = geoJsonService.importGeoData(jobParams, input);
```

Comment:
The service call is better wrapped via `runStage` in the same way line 64/65 does it, the complete is then part of `runStage`

Reply:
i'll update this